lec07-pop-models.Rmd

```diff
@@ -1,4 +1,617 @@
 ---
-title: "Qualitative population models in 2 dimensions"
+title: "Population models in two dimensions and numerically solving equations in R"
+author: "Madeleine Bonsma-Fisher"
+output: html_document
 ---
 
+> ### Learning objectives
+>
+> - Review differential equations and one-dimensional phase portraits
+> - Numerically solve differential equations in R using Euler's method and R's ODE solver
+> - Qualitatively analyze two-dimensional models by hand and in R
+
+> ### Lesson outline
+> 
+> Total lesson time: 2 hours
+> 
+> - 
+
+> ### Setup
+> 
+> - `install.packages('deSolve')`
+> - `install.packages('ggplot2')` (or `tidyverse`)
+
+
+## Recap and some tips
+
+   - You can include math in R notebooks using [LaTeX syntax](https://en.wikibooks.org/wiki/LaTeX/Mathematics). 
+
+     `$$\frac{dN}{dt} = r N (1 - N/K)$$` produces 
+     $$\frac{dN}{dt} = r N (1 - N/K)$$
+
+   - **Differential equation**: an equation that describes how a function of variables is related to
+derivatives of those variables. 
+
+   - Drawing **phase portraits** in one dimension:
+
+    ![Logistic equation simulation](https://github.com/UofTCoders/rcourse/raw/master/logistic.gif)
+
+       - Fixed points: values of $N$ at which $dN/dt$, the rate of change of $N$, is $0$. 
+       To find fixed points, plot $dN/dt$ vs. $N$ and find the place(s) where it crosses the $x$ axis. 
+       - Stability: if you start at some $N$ close to the fixed point but not exactly on it, 
+       will you go towards (stable) or away (unstable) from the fixed point? The sign of $dN/dt$ on
+       either side of a fixed point tells you whether $N$ will increase or decrease in that area. 
+       Draw an arrow to the right if $dN/dt$ is positive, and draw and arrow to the left if $dN/dt$ is negative. 
+
+### Challenge
+
+Find the fixed point(s) of the following differential equation.
+
+   - Plot $\frac{dN}{dt}$ vs. $N$ for $N$ ranging from $0$ to $10$.
+   - Sketch a phase portrait. Are the fixed point(s) stable or unstable?
+
+$$\frac{dN}{dt} = \text{sin}(N)$$
+<!--
+```{r}
+library(ggplot2)
+
+N_seq = seq(0, 10, by = 0.1)
+
+dN_dt = sin(N_seq)
+
+qplot(N_seq, dN_dt) +
+  geom_line(aes(x = N_seq, y = 0))
+```
+-->
+
+You can think of $dN/dt$ as a *velocity*: it tells you how fast $N$ is changing and in what
+direction. If we start a **trajectory** at some initial $N_0$, $dN/dt$ dictates whether $N$ will
+increase, decrease, or stay the same. If $N_0$ happens to be at a fixed point, $N$ will remain at
+$N_0$ for all time. 
+
+### Challenge (time permitting)
+
+
+Find the fixed point(s) of the following differential equation.
+
+   - Plot $\frac{dx}{dt}$ vs. $x$ for $x$ ranging from $-2$ to $2$ in $0.1$ increments.
+   - Sketch a phase portrait. Are the fixed point(s) stable or unstable?
+
+$$\frac{dx}{dt} = x^2 - 1$$
```

Comment:
I'd suggest this equation be moved to below where is says "of the following equation" (line 75)